### PR TITLE
Add headers to pass as additional headers to JWT

### PR DESCRIPTION
**Pull request by gimlet2**
That might be useful to provide `kid` for JWK

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@gimlet2 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=1436)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=1436) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@gimlet2 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=1436)!

File: spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/store/JwtAccessTokenConverter.java [hunk 1]
```diff
@@ -76,6 +76,8 @@
 	private String signingKey = verifierKey;
 
 	private SignatureVerifier verifier;
+	
+	private Map<String, String> headers = new LinkedHashMap<String, String>();
```
Comment (jgrandja) [inline spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/store/JwtAccessTokenConverter.java:80]: `JwtAccessTokenConverter` is **not thread-safe** so `headers` attribute cannot be added here. If you still require the ability to add custom headers to the JWT then please re-work the solution and submit a new PR that includes tests.

**Comment by jgrandja** [comment]
Thank you for the PR @gimlet2 but the enhancement provided has an issue. Please see my [comment](https://github.com/spring-projects/spring-security-oauth/pull/1436#discussion_r404385984).
Closing this PR but feel free to submit a new one with a revised enhancement.

**Pull request closed without merging**